contracts/core/Dispatcher.sol

```diff
@@ -86,9 +86,18 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         L1Header calldata l1header,
         OpL2StateProof calldata proof,
         uint256 height,
-        uint256 appHash
+        uint256 appHash,
+        string calldata connection
     ) external returns (uint256 fraudProofEndTime, bool ended) {
-        return lightClient.addOpConsensusState(l1header, proof, height, appHash);
+        return _getLightClientFromConnection(connection).addOpConsensusState(l1header, proof, height, appHash);
+    }
+
+    function setNewConnection(string calldata connection, LightClient lightClient) external onlyOwner {
```

Comment:
nit: add comment here the func can 1. set new client for a new connection; or 2. update an existing connection's to a new client.
a name like `setClientForConnection` feels more natural?

Reply:
added!